### PR TITLE
Fix alternatives module

### LINKED ISSUE #4803: 5.1.0 breaks `alternatives` on RHEL 8
### Summary

When I try to run the below task, I receive a usage error from `update-alternatives`. Prior to 5.1.0, this works as expected. It looks like the `--slave` argument is missing a required value (`--slave unversioned-python /usr/bin/python3` compared to `--slave <slave_link> <slave_name> <slave_path>`). Downgrading to `5.0.2` is a workaround.

### Issue Type

Bug Report

### Component Name

alternatives

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.12.6]
  config file = /tmp/y/ansible.cfg
  configured module search path = ['/var/home/bpratt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /tmp/y/venv/lib64/python3.10/site-packages/ansible
  ansible collection location = /var/home/bpratt/.ansible/collections:/usr/share/ansible/collections
  executable location = /tmp/y/venv/bin/ansible-playbook
  python version = 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 12.0.1 20220308 (Red Hat 12.0.1-0)]
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general

# /var/home/bpratt/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.1.0

# /tmp/y/venv/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 4.8.2

# /tmp/y/venv/lib64/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 4.8.2

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

RHEL 8

### Steps to Reproduce

```yaml (paste below)
    - community.general.alternatives:
        name: python
        path: /usr/bin/python3
        link: /usr/bin/python
```

### Expected Results

`{"changed": true}`

### Actual Results

```console (paste below)
TASK [community.general.alternatives] *******************************************************************************************************************************************
task path: /tmp/xyz/playbook.yml:4
redirecting (type: modules) community.general.alternatives to community.general.system.alternatives
fatal: [10.0.101.1]: FAILED! => changed=false
  ansible_facts:
    discovered_interpreter_python: /usr/libexec/platform-python
  cmd: /usr/sbin/update-alternatives --install /usr/bin/python python /usr/bin/python3 50 --slave unversioned-python /usr/bin/python3 --slave unversioned-python-man /usr/share/man/man1/python3.1.gz
  msg: ''
  rc: 2
  stderr: ''
  stderr_lines: <omitted>
  stdout: |-
    alternatives version 1.19.1 - Copyright (C) 2001 Red Hat, Inc.
    This may be freely redistributed under the terms of the GNU Public License.

    usage: alternatives --install <link> <name> <path> <priority>
                        [--initscript <service>]
                        [--family <family>]
                        [--slave <slave_link> <slave_name> <slave_path>]*
           alternatives --remove <name> <path>
           alternatives --auto <name>
           alternatives --config <name>
           alternatives --display <name>
           alternatives --set <name> <path>
           alternatives --list
           alternatives --remove-all <name>
           alternatives --add-slave <name> <path> <slave_link> <slave_name> <slave_path>
           alternatives --remove-slave <name> <path> <slave_name>

    common options: --verbose --test --help --usage --version --keep-missing --keep-foreign
                    --altdir <directory> --admindir <directory>
  stdout_lines: <omitted>
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### LINKED ISSUE #4804: Alternatives subcommand is missing link if not passed explicitly
### Summary

When I try to use the alternatives module from collection version 5.1.0, it seems that the new [subcommands feature](https://github.com/ansible-collections/community.general/pull/4654) is not behaving correctly on Red Hat systems, when subcommands are already in place.
Please see the details below.

### Issue Type

Bug Report

### Component Name

alternatives

### Ansible Version

```bash
ansible --version
ansible [core 2.12.4]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.5 (default, Nov 23 2021, 15:27:38) [GCC 9.3.0]
  jinja version = 3.1.1
  libyaml = True
```

### Community.general Version

```bash
ansible-galaxy collection list community.general

# /usr/local/lib/python3.9/dist-packages/ansible_collections
Collection        Version
----------------- -------
community.general 4.7.0  

# /root/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.1.0
```

### Configuration

```shell
ansible-config dump --only-changed
ANSIBLE_FORCE_COLOR(/azp/_work/6/s/lz/project/ansible/ansible.cfg) = True
DEFAULT_TRANSPORT(/azp/_work/6/s/lz/project/ansible/ansible.cfg) = ssh
INVENTORY_ENABLED(/azp/_work/6/s/lz/project/ansible/ansible.cfg) = ['host_list', 'script', 'auto', 'yaml', 'ini', 'azure_rm']
```

### OS / Environment

**Target system: Red Hat Enterprise Linux release 8.6 (Ootpa)**
Ansible runner: ubuntu:20.04 / Container instance in Azure

### Steps to Reproduce

```yaml
- name: Add python to alternatives system
  community.general.alternatives:
    name: python
    path: /usr/bin/python3
    link: /usr/bin/python
```
Have python3 installed:
```bash
alternatives --display python
python - status is manual.
 link currently points to /usr/bin/python3
/usr/libexec/no-python - priority 404
 slave unversioned-python: (null)
 slave unversioned-python-man: /usr/share/man/man1/unversioned-python.1.gz
/usr/bin/python3 - priority 300
 slave unversioned-python: /usr/bin/python3
 slave unversioned-python-man: /usr/share/man/man1/python3.1.gz
Current `best' version is /usr/libexec/no-python.
```

### Expected Results

I expected it would set python3 as the default python executable and any existing subcommand will be considered accordingly.

### Actual Results

Since collection version 5.1.0 the Ansible task will fail, as slave parameter are missing the slave-link argument:
```console (paste below)
fatal: [server]: FAILED! => {"changed": false, "cmd": "/sbin/update-alternatives --install /usr/bin/python python /usr/bin/python3 50 --slave unversioned-python /usr/bin/python3 --slave unversioned-python-man /usr/share/man/man1/python3.1.gz", "msg": "", "rc": 2, "stderr": "", "stderr_lines": [], "stdout": "alternatives version 1.19.1 - Copyright (C) 2001 Red Hat, Inc.\nThis may be freely redistributed under the terms of the GNU Public License.\n\nusage: alternatives --install <link> <name> <path> <priority>\n                    [--initscript <service>]\n                    [--family <family>]\n                    [--slave <slave_link> <slave_name> <slave_path>]*\n       alternatives --remove <name> <path>\n       alternatives --auto <name>\n       alternatives --config <name>\n       alternatives --display <name>\n       alternatives --set <name> <path>\n       alternatives --list\n       alternatives --remove-all <name>\n       alternatives --add-slave <name> <path> <slave_link> <slave_name> <slave_path>\n       alternatives --remove-slave <name> <path> <slave_name>\n\ncommon options: --verbose --test --help --usage --version --keep-missing --keep-foreign\n                --altdir <directory> --admindir <directory>\n", "stdout_lines": ["alternatives version 1.19.1 - Copyright (C) 2001 Red Hat, Inc.", "This may be freely redistributed under the terms of the GNU Public License.", "", "usage: alternatives --install <link> <name> <path> <priority>", "                    [--initscript <service>]", "                    [--family <family>]", "                    [--slave <slave_link> <slave_name> <slave_path>]*", "       alternatives --remove <name> <path>", "       alternatives --auto <name>", "       alternatives --config <name>", "       alternatives --display <name>", "       alternatives --set <name> <path>", "       alternatives --list", "       alternatives --remove-all <name>", "       alternatives --add-slave <name> <path> <slave_link> <slave_name> <slave_path>", "       alternatives --remove-slave <name> <path> <slave_name>", "", "common options: --verbose --test --help --usage --version --keep-missing --keep-foreign", "                --altdir <directory> --admindir <directory>"]}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #4803, fixes #4804.

CC @jiuka

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
alternatives

### CODE REVIEW

File: plugins/modules/system/alternatives.py [hunk 1]
```diff
@@ -77,6 +77,7 @@
         description:
           - The path to the symbolic link that should point to the real subcommand executable.
         type: path
+        required: true
```
Comment (felixfontein) [inline plugins/modules/system/alternatives.py:80]: When `link` wasn't specified, the module generated an invalid argument for `alternatives` and failed. So this change shouldn't break anything that was working.

**Comment by pilou-** [comment]
> I did some quick tests with RHEL 8.5 and python3, and it seems to work fine.

Should not this have been caught by the [`alternatives` integration tests](https://dev.azure.com/ansible/community.general/_build/results?buildId=43665&view=logs&j=458389df-61af-5f74-279d-6c277dd49fc8&t=bedc0318-f287-5c42-bf8a-d498ade199ab) run within the CI when `subcommands` was added (#4654)?

**Comment by felixfontein** [comment]
The tests do not cover this since they never switch between alternatives when subcommands are present.

**Comment by felixfontein** [comment]
The tests should now cover that. I also added some more debug output so that when they start failing in the future, its (hopefully) easier to figure out what goes wrong.

**Comment by felixfontein** [comment]
@pilou- thanks a lot for reviewing and testing this!

**Pull request merged**